Analytics/AnalyticsLogging/AnalyticsLogger.swift

```diff
@@ -106,10 +110,26 @@ class AnalyticsLogger {
             case .success(let success):
                 completion(.success(success.data?.putAnalyticEvent.success ?? false))
             case .failure(let error):
-                completion(.failure(error))
+                self.attemptToRetryWhenFailed(
+                    with: error,
+                    event: event,
+                    completion: completion)
             }
         }
     }
+
+    private func attemptToRetryWhenFailed(
+        with error: Error,
+        event: AnalyticEvent,
+        completion: @escaping (Result<Bool, Error>) -> Void
+    ) {
+        if attempts < maxAttempts {
+            attempts += 1
+            performPutAnalyticEventMutation(event: event, completion: completion)
+        } else {
+            completion(.failure(error))
+        }
+    }
 }
```

Comment:
The reason that we need to have a retry function when failed is because app is returning this error when opening push notification. So we cannot send `pushOpened` tracking event to BE successfully.
```
The network connection was lost." UserInfo={_kCFStreamErrorCodeKey=-4, NSUnderlyingError=0x280d72ac0 {Error Domain=kCFErrorDomainCFNetwork Code=-1005 "(null)" UserInfo={NSErrorPeerAddressKey=<CFData 0x28219edf0 [0x1c52db5e0]>{length = 16, capacity = 16, bytes = 0x10022382ac140a020000000000000000}, _kCFStreamErrorCodeKey=-4, _kCFStreamErrorDomainKey=4}}, _NSURLErrorFailingURLSessionTaskErrorKey=LocalDataTask <BAE6924B-771C-431E-AE71-F10CC8C7D118>.<7>, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDataTask <BAE6924B-771C-431E-AE71-F10CC8C7D118>.<7>"
), NSLocalizedDescription=The network connection was lost., NSErrorFailingURLStringKey=https://staging-graphql-eu.realifetech.com/, NSErrorFailingURLKey=https://staging-graphql-eu.realifetech.com/, _kCFStreamErrorDomainKey=4})
```
Based on Apple's suggestion, I think it's safe for us to perform a retry when we see that error. So I've tried this and the mutation works successfully after the retry.
https://developer.apple.com/library/archive/qa/qa1941/_index.html#//apple_ref/doc/uid/DTS40017602